spec/requests/profile_spec.rb

```diff
@@ -0,0 +1,164 @@
+            stories = [build_stubbed(:moment), build_stubbed(:strategy)]
+            allow_any_instance_of(ProfileController)
+              .to receive(:get_stories).and_return(stories)
+            get profile_index_path, params: {uid: user1.uid}
+            expect(assigns(:stories)).to match_array(stories)
+          end
+        end
+      end
+
+      context "when profile does not belong to user" do
+        let(:profile_owner) { build_stubbed(:user2) }
+
+        before do
+          allow(User).to receive(:find_by).and_return(profile_owner)
+          get profile_index_path, params: {uid: profile_owner.uid}
+        end
+
+        it "does not assign stories" do
+          expect(assigns(:stories)).to be_nil
+        end
+
+        it "assigns profile instance to profile owner" do
+          expect(assigns(:profile)).to eq(profile_owner)
+        end
+      end
+    end
+  end
+
+  describe "#data" do
+    let!(:user) { create(:user) }
+    let!(:moment) { create(:moment, user: user) }
+
+    context "when the user is signed in" do
+      before { sign_in user }
+      before { get data_profile_index_path, params: {page: 1, uid: user.uid}, headers: {"ACCEPT" => "application/json"} }
+
+      it "returns a response with the correct path" do
+        expect(JSON.parse(response.body)["data"].first["link"]).to eq moment_path(moment)
+      end
+    end
+
+    context "when the user is not signed in" do
+      before { get data_profile_index_path params: {page: 1, uid: user.uid}, headers: {"ACCEPT" => "application/json"} }
+
+      it "returns a no_content status" do
+        expect(response).to have_http_status(:no_content)
+      end
+    end
+  end
+
+  describe "#add_ban" do
+    let(:user2) { create(:user2) }
+
+    before(:each) { Devise.mailer.deliveries.clear }
+
+    context "when admin does not exist" do
+      let(:user1) { create(:user1) }
+
+      it "cannot ban user" do
+        sign_in user1
+        post add_ban_profile_index_path, params: {user_id: user2.id}
+        expect(response.status).to eq(204)
+        expect(Devise.mailer.deliveries.count).to eq(0)
+      end
+    end
+
+    context "when admin exists" do
+      let(:user1) { create(:user1, admin: true) }
+
+      context "when user exists" do
+        it "bans the user" do
+          sign_in user1
+          post add_ban_profile_index_path, params: {user_id: user2.id}
+          expect(response).to redirect_to(admin_dashboard_path)
+          expect(flash[:notice]).to eq("#{user2.name} has been banned")
+          expect(Devise.mailer.deliveries.count).to eq(1)
+        end
+      end
+
+      context "when user does not exist" do
+        it "does not ban the user" do
+          sign_in user1
+          post add_ban_profile_index_path, params: {user_id: -1}
+          expect(response).to redirect_to(admin_dashboard_path)
+          expect(flash[:alert]).to eq("Could not ban -1")
+          expect(Devise.mailer.deliveries.count).to eq(0)
```

Comment:
```suggestion
          sign_in user1
          expect { post add_ban_profile_index_path, params: {user_id: -1} }
            .not_to change(Devise.mailer.deliveries, :count).from(0)
          expect(response).to redirect_to(admin_dashboard_path)
          expect(flash[:alert]).to eq("Could not ban -1")
```

Since we're making sure that the mailer count isn't changing to avoid false positives.